src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,110 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+
+    private Node<K, V>[] buckets;
+    private int capacity;
+    private int threshold;
+    private int size;
+
+    public MyHashMap() {
+        this.capacity = DEFAULT_CAPACITY;
+        this.buckets = new Node[DEFAULT_CAPACITY];
+        this.size = 0;
+        this.threshold = (int) (capacity * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size >= threshold) {
+            resize();
+        }
+
+        int index = getIndex(key);
+        Node<K, V> current = buckets[index];
+
+        while (current != null) {
+            if (isKeysEqual(key, current.key)) {
+                current.value = value;
+                return;
+            }
+            current = current.next;
+        }
 
+        Node<K, V> newNode = new Node<>(key == null ? 0 : key.hashCode(), key, value);
+        newNode.next = buckets[index];
+        buckets[index] = newNode;
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        Node<K, V> current = buckets[index];
+
+        while (current != null) {
+            if (isKeysEqual(key, current.key)) {
```

Comment:
```suggestion
            if (key == node.key || key != null && key.equals(node.key)) {
```

Or you can use you method isKeysEqual but refactor it